src/Orchard.Web/Modules/Orchard.OpenId/Views/Admin/Edit.cshtml

```diff
@@ -0,0 +1,81 @@
+    @Html.AntiForgeryToken()
+    <h4>@T["Edit OpenId Application"]</h4>
+    <hr />
+    <input type="hidden" asp-for="Id" />
+    <div asp-validation-summary="All" class="text-danger"></div>
+    <div class="form-group">
+        <label asp-for="ClientId" class="col-md-2 control-label"></label>
+        <div class="col-md-10">
+            <input asp-for="ClientId" class="form-control" />
+            <span asp-validation-for="ClientId" class="text-danger"></span>
+        </div>
+    </div>
+    <div class="form-group">
+        <label asp-for="DisplayName" class="col-md-2 control-label"></label>
+        <div class="col-md-10">
+            <input asp-for="DisplayName" class="form-control" />
+            <span asp-validation-for="DisplayName" class="text-danger"></span>
+        </div>
+    </div>
+    <div class="form-group">
+        <label asp-for="RedirectUri" class="col-md-2 control-label"></label>
+        <div class="col-md-10">
+            <input asp-for="RedirectUri" class="form-control" />
+            <span asp-validation-for="RedirectUri" class="text-danger"></span>
+        </div>
+    </div>
+    <div class="form-group">
+        <label asp-for="LogoutRedirectUri" class="col-md-2 control-label"></label>
+        <div class="col-md-10">
+            <input asp-for="LogoutRedirectUri" class="form-control" />
+            <span asp-validation-for="LogoutRedirectUri" class="text-danger"></span>
+        </div>
+    </div>
+    <div class="form-group">
+        <label asp-for="Type" class="col-md-2 control-label"></label>
+        <div class="col-md-10">
+            <select asp-for="Type" class="form-control">
+                <option value="@ClientType.Confidential">@ClientType.Confidential.ToString()</option>
+                <option value="@ClientType.Public">@ClientType.Public.ToString()</option>
+            </select>
+            <span asp-validation-for="Type" class="text-danger"></span>
+        </div>
+    </div>
+    @{
+        var rolesClass = Model.Type == ClientType.Public ? "form-group hidden" : "form-group";
+    }
+    <div class=@rolesClass id="RoleGroup" name="RoleGroup">
+        <label class="col-md-2 control-label">@T["Roles"]</label>
+        <div class="col-md-10">
+            <ul class="list-group">
+                @for (var i = 0; i < Model.RoleEntries.Count; i++)
+                {
+                    <li class="list-group-item">
+                        <input asp-for="RoleEntries[i].Selected" type="checkbox">
+                        <input asp-for="RoleEntries[i].Name" type="hidden">
+                        @Model.RoleEntries[i].Name
+                    </li>
+                }
+            </ul>
+        </div>
+    </div>
+    <div class="form-group">
+        <div class="col-md-offset-2 col-md-10">
+            <button type="submit" class="btn btn-default">@T["Save"]</button>
+        </div>
+    </div>
+</form>
+<script at="Foot">
```

Comment:
This doesn't work, the script is missing from the resulting html page.
If I remove at="Foot" attribute then the script appears at output but before JQuery link. So, it fails because it depends on JQuery.